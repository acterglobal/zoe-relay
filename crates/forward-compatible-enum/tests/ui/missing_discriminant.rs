use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
pub enum MissingDiscriminant {
    Variant1,  // Missing #[discriminant(N)]
    #[discriminant(1)]
    Variant2,
    Unknown { discriminant: u32, data: Vec<u8> },
}