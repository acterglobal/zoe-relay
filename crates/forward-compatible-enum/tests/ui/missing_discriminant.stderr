error: all variants must have #[discriminant(N)] attribute
 --> tests/ui/missing_discriminant.rs:3:10
  |
3 | #[derive(ForwardCompatibleEnum)]
  |          ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `ForwardCompatibleEnum` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/missing_discriminant.rs:9:2
  |
9 | }
  |  ^ consider adding a `main` function to `$DIR/tests/ui/missing_discriminant.rs`
